code/modules/surgery/bodyparts/bodyparts.dm

```diff
@@ -107,52 +108,54 @@
 //Applies brute and burn damage to the organ. Returns 1 if the damage-icon states changed at all.
 //Damage will not exceed max_damage using this proc
 //Cannot apply negative damage
-/obj/item/bodypart/proc/receive_damage(brute, burn, updating_health = 1)
+/obj/item/bodypart/proc/receive_damage(brute = 0, burn = 0, stamina = 0, updating_health = TRUE)
 	if(owner && (owner.status_flags & GODMODE))
-		return 0	//godmode
+		return FALSE	//godmode
 	var/dmg_mlt = CONFIG_GET(number/damage_multiplier)
 	brute = max(brute * dmg_mlt, 0)
 	burn = max(burn * dmg_mlt, 0)
-
-
+	stamina = max(stamina * dmg_mlt, 0)
 	if(status == BODYPART_ROBOTIC) //This makes robolimbs not damageable by chems and makes it stronger
 		brute = max(0, brute - 5)
 		burn = max(0, burn - 4)
+		//No stamina scaling.. for now..
+
+	if(!brute && !burn && !stamina)
+		return FALSE
 
 	switch(animal_origin)
 		if(ALIEN_BODYPART,LARVA_BODYPART) //aliens take double burn
 			burn *= 2
 
 	var/can_inflict = max_damage - (brute_dam + burn_dam)
-	if(!can_inflict)
-		return 0
+	if(can_inflict <= 0)
+		return FALSE
+
+	var/total_damage = brute + burn
+
+	if(total_damage > can_inflict)
+		var/excess = total_damage - can_inflict
+		brute = brute * (excess / total_damage)
+		burn = burn * (excess / total_damage)
+
+	brute_dam += brute
+	burn_dam += burn
+
+	//We've dealt the physical damages, if there's room lets apply the stamina damage.
+	var/current_damage = brute_dam + burn_dam + stamina_dam		//This time around, count stamina loss too.
+	var/available_damage = max_damage - current_damage
+	stamina_dam += CLAMP(stamina, 0, available_damage)
 
-	if((brute + burn) < can_inflict)
-		brute_dam	+= brute
-		burn_dam	+= burn
-	else
-		if(brute > 0)
-			if(burn > 0)
-				brute	= round( (brute/(brute+burn)) * can_inflict, 1 )
-				burn	= can_inflict - brute	//gets whatever damage is left over
-				brute_dam	+= brute
-				burn_dam	+= burn
-			else
-				brute_dam	+= can_inflict
-		else
-			if(burn > 0)
-				burn_dam	+= can_inflict
-			else
-				return 0
 	if(owner && updating_health)
 		owner.updatehealth()
+		if(stamina)
+			owner.update_stamina()
```

Comment:
Again. Make sure update_stamina() ALWAYS gets called regardless of updating health or not.